src/shogun/evaluation/TimeSeriesSplitting.cpp

```diff
@@ -84,7 +83,7 @@ void CTimeSeriesSplitting::build_subsets()
 		SG_UNREF(current);
 	}
 
-	m_subset_indices->shuffle(m_rng);
+	m_subset_indices->shuffle(this);
```

Comment:
the array should have its own rng with this new design imo.
But this is minor  ...